### PR TITLE
Initial docs

### PR DESCRIPTION
**Pull request by woutslakhorst**
Added basic documentation on how the discovery service functions and how it can be setup. More will follow when the code has been completed.
    
    Added the OpenSSL configuration files for generating the correct keys and certificates for local development.

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -1,10 +1,13 @@
-.. Nuts discovery documentation master file, created by
-   sphinx-quickstart on Fri Feb  8 14:20:59 2019.
-   You can adapt this file completely to your liking, but it should at least
-   contain the root `toctree` directive.
+.. _nuts-discovery:
 
-Welcome to Nuts discovery's documentation!
-==========================================
+Nuts Discovery
+==============
```
Comment (stevenvegt) [inline docs/index.rst]: This might be a nice place to explain what the discovery service is.

File: docs/pages/api.rst [hunk 2]
```diff
@@ -0,0 +1,121 @@
+.. _nuts-discovery-api:
+
+***
+Api
+***
+
+The *Nuts Discovery Service* consists of 2 api's: the network map and certificare api's. The certificate api is in the initial setup phase of the Corda node. To publish the node details with an electronic signature and retrieve the signed certificate from the *Nuts Discovery Service*. The network map api is used to retrieve details about all the nodes that are connected to the network.
```
Comment (stevenvegt) [inline docs/pages/api.rst]: Typo: certificare

Comment (stevenvegt) [inline docs/pages/api.rst]: > The certificate api is in the initial setup phase of the Corda node.

Is wat, used, only used?

File: docs/pages/api.rst [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+Certificate API
+===============
+
+The *Certificate API* is part of the Corda node but it's seems to be lacking documentation. After reverse engineering, 2 requests could be identified. The initation process for a Corda node can be started with:
+::
+
+    java -jar corda.jar initial-registration -p changeit
```
Comment (stevenvegt) [inline docs/pages/api.rst]: `changeit` is apparently the password. Maybe put it like this `<password>`

File: docs/pages/api.rst [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+where the given password is the password of the root truststore. More info on how to start and configure a node can be found in :ref:`node setup <nuts-discovery-setup>`.
```
Comment (stevenvegt) [inline docs/pages/api.rst]: The root truststore is a Corda thing right? If so, prefixing it with Corda like `the Corda root truststore` can clarify things.

File: docs/pages/api.rst [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+
+When started, the Corda node will create a couple of keystore as seen `here <https://docs.corda.net/permissioning.html#certificate-hierarchy>`_. The node will send a certificate request to the configured *Doorman* url followed by /certificate. The return value will be an identifier which the node will use to poll the service. When the service has signed the request, the certificate is then downloaded and put in the keystore. Below are the details of the two services.
```
Comment (stevenvegt) [inline docs/pages/api.rst]: couple of Keystore -> plural -> keystores

Comment (stevenvegt) [inline docs/pages/api.rst]: What was that Doorman thing again? :)

File: docs/pages/api.rst [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+
+.. http:post:: /certificate
+
+   Receiving endpoint for a certificate request (CSR). The body must be in PKCS10 byte format. The response is a plain text response with an identifier which can be used in the GET call.
+
+   :reqheader Platform-Version: the Corda platform version (4)
+   :reqheader Client-Version: todo
+   :reqheader Private-Network-Map: todo
```
Comment (stevenvegt) [inline docs/pages/api.rst]: Should the user provide a content type `text/plain`?

File: docs/pages/api.rst [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+
+.. http:get:: /(request_id)
+
+   Retrieve the signed certificate which was requesting using the POST method. `request_id` is the returned identifier from the POST request. The body contains a zip file with 3 files: **cordaclientca.cer**, **cordaintermediateca.cer** and **cordarootca.cer** (in this order). Each file is the ASN.1 DER encoding of a X.509 certificate.
+
+   :resheader Content-Disposition: (attachment; filename="certificates.zip")
+
+   :statuscode 200: Certificate request has been signed, the body consists of a zip file with a list of certificates.
+   :statuscode 400: something went wrong.
+   :statuscode 404: Unknown identifier or request hasn't been signed yet.
```
Comment (stevenvegt) [inline docs/pages/api.rst]: So, this endpoint should be polled. If the request has not been signed you get a 404?
If the request has been signed, how often can the `request_id` be used?

File: docs/pages/keys.rst [hunk 8]
```diff
@@ -0,0 +1,9 @@
+.. _nuts-discovery-keys:
+
+*********************
+Keys and certificates
+*********************
+
+The figure at https://docs.corda.net/network-map.html shows the CA structure of Corda nodes. The *Nuts Discovery Service* represents the *security zone* in that figure. To startup the service, a few keys need to be generated and configured. More info on this can be found in :ref:`nuts-discovery-setup`. Only the root CA is configured at each node as trusted anchor. Both the Doorman and Network Map CA are trusted because they are signed by the root. The CertRole for each CA is important, to prevent mistakes configuration files for generating these certificates can be found in this repository.
```
Comment (stevenvegt) [inline docs/pages/keys.rst]: This does not link to a figure.
Also, the Corda docs are still pointing to v3. Should we point to Corda docs head (v4)?

File: docs/pages/keys.rst [hunk 9]
```diff
@@ -0,0 +1,9 @@
...
+
+All mentioned keys and certificates are only used for the Corda part of the Nuts network. Communication between care providers use different certificates. This ensures that only the right parties can issue/revoke certificates based on the responsibilities they have. The *Nuts Discovery Service* can only determine who joins the network. It cannot access or manipulate any data. The root key should not be present on any production environments. It can even be controlled by a different part of the Nuts community to ensure decentralization of control.
```
Comment (stevenvegt) [inline docs/pages/keys.rst]: Can you display this in an _info_ or _warning_ block? It feels like a standalone hint.

File: docs/index.rst (hunk 1 above)
Comment (woutslakhorst) [reply docs/index.rst]: The general explanation is in the main docs. We can duplicate that?

File: docs/pages/keys.rst (hunk 8 above)
Comment (woutslakhorst) [reply docs/pages/keys.rst]: Problem with linking to their docs is that it'll break when they change their docs...

File: docs/pages/api.rst (hunk 6 above)
Comment (woutslakhorst) [reply docs/pages/api.rst]: is done by the Corda client

**Pull request merged**